api/src/models/project-survey-attachments.ts

```diff
@@ -10,25 +10,34 @@ const defaultLog = getLogger('models/project-survey-attachments');
  * @class GetAttachmentsData
  */
 export class GetAttachmentsData {
-  attachmentsList: any[];
-  reportAttachmentsList: any[];
-
-  constructor(attachmentsData?: any, reportAttachmentsData?: any) {
-    defaultLog.debug({ label: 'GetAttachmentsData', message: 'params', attachmentsData });
+  constructor() {
+    defaultLog.debug({ label: 'GetAttachmentsData', message: 'params' });
+  }
 
-    const mapAttachment = (item: any) => {
+  static async buildAttachmentsData(attachmentsData: any[], getSupplementaryData: (attachmentId: number) => any) {
+    const mapAttachment = (attachment: any, supplementaryData: any) => {
       return {
-        id: item.id,
-        fileName: item.file_name,
-        fileType: item.file_type || 'Report',
-        lastModified: moment(item.update_date || item.create_date).toISOString(),
-        size: item.file_size,
-        status: item.status
+        id: attachment.survey_attachment_id || attachment.survey_report_attachment_id,
+        fileName: attachment.file_name,
+        fileType: attachment.file_type || 'Report',
+        lastModified: moment(attachment.update_date || attachment.create_date).toISOString(),
+        size: attachment.file_size,
+        status: attachment.status,
+        supplementaryAttachmentData: supplementaryData
       };
     };
 
-    this.attachmentsList = (attachmentsData?.length && attachmentsData.map(mapAttachment)) || [];
-    this.reportAttachmentsList = (reportAttachmentsData?.length && reportAttachmentsData.map(mapAttachment)) || [];
+    const attachmentList: any[] = [];
+
+    attachmentsData.map(async (attachment: any) => {
+      const supplementaryData = await getSupplementaryData(
+        attachment.survey_attachment_id || attachment.survey_report_attachment_id
+      );
+
+      attachmentList.push(mapAttachment(attachment, supplementaryData));
+    });
+
+    return attachmentList;
```

Comment:
I don't think fetching supplemental data in the model is the right place for it.
It would be better to make these calls in a service function(s).

It would be ideal to remove the model altogether, but that is maybe an issue for a separate tech debt ticket.